dataland-frontend/src/components/resources/frameworkDataSearch/TwoLayerDataTable.vue

```diff
@@ -246,47 +248,17 @@ export default defineComponent({
      * @returns true if the string is 'Yes' or 'No'
      */
     isYesNo(value: string) {
-      return Object.values(YesNo).includes(value);
+      return Object.values(YesNo).includes(value as YesNo);
     },
     /**
-     * Opens a modal to display a table with the provided list of production sites
-     * @param listOfValues An array consisting of the data to display
-     * @param modalTitle The title for the modal, which is derived from the key of the KPI
-     * @param kpiKey the key of the KPI used to determine the type of Subtable that needs to be displayed
-     * @param kpiFormFieldComponent determine whether a specific component should be used to render data
+     * @param content the array value from the field
+     * @returns an array with undefined and null values filtered out
      */
-    openModalAndDisplayValuesInSubTable(
-      listOfValues: [],
-      modalTitle: string,
-      kpiKey: string,
-      kpiFormFieldComponent = "DetailsCompanyDataTable",
-    ) {
-      let kpiDataComponent:
-        | typeof DetailsCompanyDataTable
-        | typeof AlignedActivitiesDataTable
-        | typeof NonAlignedActivitiesDataTable;
-      if (kpiFormFieldComponent === "AlignedActivitiesDataTable") {
-        kpiDataComponent = AlignedActivitiesDataTable;
-      } else if (kpiFormFieldComponent === "NonAlignedActivitiesDataTable") {
-        kpiDataComponent = NonAlignedActivitiesDataTable;
-      } else {
-        kpiDataComponent = DetailsCompanyDataTable;
+    filterStringArray(content: unknown[]): string {
+      if (Array.isArray(content) && content.length > 0) {
```

Comment:
I know in terms of temporal decoupling it makes sense to recheck if content is an array here again. But this is a double check, now. We already check this before it is called from the HTML part. Also this gets kind of obsolete if the type of content is already an array type.

Reply:
this function no longer exists